drivers/config.go

```
@@ -0,0 +1,74 @@
+
+	return fmt.Sprintf("postgresql://%s:%s@%s/%s", s.Username, url.QueryEscape(s.Secret), s.Address, s.Database)
+}
+
+// Looks up CNAME record to get an RDS instance identifier and builds an IAM auth token, returning a connection string
+func (s DatabaseConfiguration) RDSIAMAuthConnectionString() string {
+	if cfg, err := awsConfig.LoadDefaultConfig(context.TODO()); err != nil {
+		slog.Error("AWS Config Loading Error", slog.String("err", err.Error()))
+	} else {
+		// Must use instance endpoint with IAM auth
+		host := s.Address
+		if hostCName, err := net.LookupCNAME(s.Address); err != nil {
```

golangci-lint (2.11.4)

[error] net.LookupCNAME must not be called. use (*net.Resolver).LookupCNAME with a context

(noctx)